### PR TITLE
Use recursive include for requirements globbing

### PR DESCRIPTION
**Pull request by paulbovbel**
Allows using the full range of requirements constraints supported by pip, and gets rid of a bunch of code from catkin_virtualenv. Win-win.

### CODE REVIEW

**Comment by paulbovbel** [comment]
Multiple requirements files can be nominally passed in via multiple `-r` flags. However, it seems that pip is unable to resolve incompatible requirements, although an issue to do proper dependency resolution in pip is in progress (https://github.com/pypa/pip/issues/988#issuecomment-360846457).

File: test_catkin_virtualenv/test/test_virtualenv_library.py [hunk 1]
```diff
@@ -23,4 +23,5 @@
 class TestVirtualenv(unittest.TestCase):
 
     def test_import_wrapt(self):
```
Comment (tappan-at-git) [inline test_catkin_virtualenv/test/test_virtualenv_library.py]: No longer imports wrapt!

File: test_catkin_virtualenv_inherit/test/test_virtualenv_library.py [hunk 2]
```diff
@@ -16,3 +16,12 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
+import importlib
+import unittest
+
+
+class TestVirtualenv(unittest.TestCase):
+
+    def test_import_wrapt(self):
```
Comment (tappan-at-git) [inline test_catkin_virtualenv_inherit/test/test_virtualenv_library.py]: ditto.

File: test_catkin_virtualenv_py3/test/test_virtualenv_library.py [hunk 3]
```diff
@@ -23,4 +23,5 @@
 class TestVirtualenv(unittest.TestCase):
 
     def test_import_wrapt(self):
```
Comment (tappan-at-git) [inline test_catkin_virtualenv_py3/test/test_virtualenv_library.py]: Hat trick.

File: test_catkin_virtualenv/test/test_virtualenv_library.py (hunk 1 above)
Comment (paulbovbel) [reply test_catkin_virtualenv/test/test_virtualenv_library.py]: sharp eye!

**Comment by paulbovbel** [comment]
I don't think this strategy is viable unless pip can resolve multiple requirements as ticketed. A smarter 'requirement merging' strategy for catkin_virtualenv might be feasible, handling multiple inline constraints.

**Pull request closed without merging**